### PR TITLE
Set up a "Hello world" project as one app with Webpack

### PR DESCRIPTION
**Pull request by Abdulwhbhussain**
Project requirements
Create a new Rails app called 'hello-rails-react'.
Use the command rails new -j webpack my_new_rails_app as described in the previous lesson's tutorial.
Verify that the jsbundling-rails gem is added to your Gemfile.
Initialize your project with Git.
Install needed node dependencies as described in the previous lesson's tutorial.
Create a babel.config.js in the root folder of your project as described in the previous lesson's tutorial.
Update webpack.config.js to include the babel loader as described in the previous lesson's tutorial.
Make sure that your project has a Postgres database set up.
Create a table for storing your messages and create 5 different greetings.
Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
Create a static view that will be the root of your app.
Create the App component with react-router and render it in your static view.
Create the Greeting component that will display a greeting. Set it up as a route in your App component.
Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
Display the random greeting in your Greeting component.

What do you think where I am Missing in setting up react. I am unable to setup react.

### CODE REVIEW

**Review by Gambit142 — COMMENTED** [review]
Hi @Abdulwhbhussain 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

- The linters for this project are not running on GitHub. Kindly ensure that the necessary linters run. 👍 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: config/routes.rb [hunk 1]
```diff
@@ -6,5 +6,6 @@
   get "up" => "rails/health#show", as: :rails_health_check
 
   # Defines the root path route ("/")
-  # root "posts#index"
+
+  root to: "messages#show"
 end
```
Comment (Qoosim) [inline config/routes.rb]: - [x] Kindly check your  root route properly. It is supposed to point to the root#index. Something like this. 

``` 
  root "root#index"
```
and there should be a route for that view.  At this moment, you don't have index for your view. This is what you have :point_down: 

view-show

### Your page after starting the app.

result


- [x] There should a route to generate random greetings in your route.rb. That, is not implemented at this moment. Kindly do that for your app to work as expected. Kindly look at bellow snippet to gain some insight


```
   resources :greetings, only: [:index], defaults: {format: 'json'} do
    collection do
      get :random_greeting
    end
  end  

```

File: app/javascript/application.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+import ReactDOM from 'react-dom';
+
+function App() {
+  return (<h1>Hello World!</h1>);
+}
+
+ReactDOM.render(
+  <App/>,
+  document.getElementById('root'),
+);
```
Comment (Qoosim) [inline app/javascript/application.js]: -  [ ]  According to the project requirements, you're expected to all everything relating to the react inside javascript folder in your app. Iniside it, you're going to setup the Greeting component, redux, store - action, reducer, and  App component. Everything that represents your frontend. Those requirements are yet to be fulfilled. Kindly attent to them. 

project-rqt

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @Abdulwhbhussain, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  Implemented some project requirements. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 1 above)
Comment (Abdulwhbhussain) [reply config/routes.rb]: Sure! I fixed it!

Comment (Abenezer-Tilahun) [reply config/routes.rb]: Fixed.

File: app/javascript/application.js (hunk 2 above)
Comment (Abenezer-Tilahun) [inline app/javascript/application.js]: - [x] As mentioned by the previous reviewer Kindly Implemented **React and Redux part** : everything relating to the react inside JavaScript folder in your app. Inside it, you're going to setup the Greeting component, redux, store - action, reducer, and App component. Everything that represents your frontend. Those requirements are yet to be fulfilled. Kindly Implemented  to them.
image

**Review by Abenezer-Tilahun — CHANGES_REQUESTED** [review]
Hi @Abdulwhbhussain 

Great work on making the changes requested by a previous reviewer 👏
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.

almost

## Required Changes ♻️
_Check the comments below the review._


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Abdulwhbhussain** [comment]
I am getting error while running this command: **rails new -j webpack new-hello-rails-react -T --database=postgresql**

jsbundler-error

So, There is no Javascript support and webpack. And It is just rendering An Empty View. It should render the Application.js - App Function. But it is not. I can work on it separately like React as a separate Frontend and Ror as Backend (API).

kindly, Can I get This Approved. I Tried Every solution. IT has to do with versioning conflicts that I am unable to resolve.

File: app/javascript/application.js (hunk 2 above)
Comment (Abdulwhbhussain) [reply app/javascript/application.js]: I am getting error while running this command: **rails new -j webpack new-hello-rails-react -T --database=postgresql**

jsbundler-error

So, There is no Javascript support and webpack.

Comment (MahmoudBakr23) [inline app/javascript/application.js]: - [x] I'm afraid I can't approve your application as mentioned by previous reviewers. Please go back to your project in the curriculum and make sure you understand every step in making react-rails in one application. Check the required gems to use to implement the functionalities. 🙏 Good luck and happy coding.

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hi @Abdulwhbhussain 😃, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but some have still not been addressed yet.

> I'd like to remind you that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

N/A

Cheers, and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/javascript/application.js (hunk 2 above)
Comment (Abdulwhbhussain) [reply app/javascript/application.js]: Worked on Requested Changes!

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @Abdulwhbhussain👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

image

## To Highlight 🎉
- Good job with Hello World project. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**